src/features/embalm/stepContent/hooks/useBundlr.ts

```diff
@@ -1,30 +1,37 @@
+import { ChunkingUploader } from '@bundlr-network/client/build/common/chunkingUploader';
 import { useToast } from '@chakra-ui/react';
 import { BigNumber } from 'ethers';
 import { formatEther } from 'ethers/lib/utils';
 import { ArweaveFileMetadata } from 'hooks/useArweaveService';
-import { chunkedUploaderFileSize } from 'lib/constants';
+import { fundStart, fundSuccess, withdrawStart, withdrawSuccess } from 'lib/utils/toast';
+import { useCallback, useContext, useEffect, useRef, useState } from 'react';
 import {
-  fundStart,
-  fundSuccess,
-  uploadStart,
-  uploadSuccess,
-  withdrawStart,
-  withdrawSuccess,
-} from 'lib/utils/toast';
-import { useCallback, useState } from 'react';
-import { fund as fundAction, setIsFunding, withdraw as withdrawAction } from 'store/bundlr/actions';
+  fund as fundAction,
+  setIsFunding,
+  setIsUploading,
+  setUploadProgress,
+  withdraw as withdrawAction,
+} from 'store/bundlr/actions';
 import { useDispatch, useSelector } from 'store/index';
+import { CreateSarcophagusContext } from '../context/CreateSarcophagusContext';
 
 export function useBundlr() {
   const dispatch = useDispatch();
   const toast = useToast();
 
   // Pull some bundlr data from store
-  const { bundlr, txId, isConnected, isFunding } = useSelector(x => x.bundlrState);
+  const { bundlr, isFunding } = useSelector(x => x.bundlrState);
 
   // Used to tell the component when to render loading circle
   const [isWithdrawing, setIsWithdrawing] = useState(false);
 
+  const [chunkedUploader, setChunkedUploader] = useState<ChunkingUploader>();
```

Comment:
Most of the updates on this file is meant to decouple uploader and event listener setup from the actual call to upload. Previous version was causing state conflicts and preventing the progress state (even in global state) from being read on the presentation layer.